cmd/cloud-controller-manager/app/options/options.go

```diff
@@ -115,6 +116,10 @@ func NewCloudControllerManagerOptions() (*CloudControllerManagerOptions, error)
 // NewDefaultComponentConfig returns cloud-controller manager configuration object.
 func NewDefaultComponentConfig(insecurePort int32) (*ccmconfig.CloudControllerManagerConfiguration, error) {
 	versioned := &ccmconfigv1alpha1.CloudControllerManagerConfiguration{}
+	// Use lease-based leader election to reduce cost.
+	// The default endpoints-leases one has already been used for couple releases.
+	versioned.LeaderElection.ResourceLock = resourcelock.LeasesResourceLock
```

Comment:
be specific about which release we switched the default to endpointslease. also, why not change the default in SetDefaults_CloudControllerManagerConfiguration instead of here?

Reply:
Added a comment about introducing in 1.17 (same below).

Modifying defaulting seems like backward-incompatible (though technically it's probably very unlikely someone will reuse it for something else...).

Should I really change to modifying the defaulting?